subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4628,28 +4534,129 @@ void lwm2m_socket_del(struct lwm2m_ctx *ctx)
 	}
 }
 
-/* LwM2M main work loop */
+static void create_notifications(sys_slist_t *observer_list,
+				 const int64_t timestamp)
+{
+	struct observe_node *obs;
+	int rc;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(observer_list, obs, node) {
+		if (obs->event_timestamp > obs->last_timestamp &&
```

Comment:
This does not match the original condition:
```
		if (obs->event_timestamp > obs->last_timestamp &&
		    (obs->min_period_sec == 0 ||
		     timestamp > obs->last_timestamp +
				MSEC_PER_SEC * obs->min_period_sec)) {
```
Note the `obs->min_period_sec == 0` check missing.

Also please keep the original comments, the code is easier to understand that way.

Reply:
Not to be snippy, but which comments do you refer to: the ones that literally repeat the condition in the if on the next line, or the ones that describe something that doesn't happen anywhere near the comment?

Reply:
Ok, I see your point, the original comments are a bit bloated indeed. I was referring to the `if` conditions, which are not very approachable IMO w/o short explainiation what exactly is being checked (especially the first one) but we can live w/o comments as well I guess.

Reply:
How about extracting the checks to seperate functions with names?

Reply:
Sound fine to me